lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -161,6 +161,61 @@ def initialize
       @prev_time = Time.now
     end
 
+    def create_k8s_client
```

Comment:
Consider moving all this code into a module to reduce the complexity of the main filter file